nix/postgres.nix

```diff
@@ -257,6 +308,9 @@ in
               # Setup config
               set -x
               cp ${configFile} "$PGDATA/postgresql.conf"
+
+              # Configure authentication
+              echo "${pgHbaConfString}" | ${pkgs.util-linux}/bin/column -t -s $'\t' > "$PGDATA/pg_hba.conf"
```

Comment:
Do we really need to create this file separately? Why not put the store path in `settings.hba_file`? See https://github.com/NixOS/nixpkgs/blob/ee5cc38432031b66e7fe395b14235eeb4b2b0d6e/nixos/modules/services/databases/postgresql.nix#L443

Reply:
I was thinking of using `writeText` for this but wasn't really sure how I can use `column` tool to make the configuration file readable.

Reply:
You need `pkgs.runCommand` I think (then write to `$out` inside).

Reply:
I have updated the PR to use `pkgs.runCommand` and I am using `settings.hba_file`. This PR should be good to be merged. I think it will be good to merge after adding a nixos test to verify if the `hba_file` being used by postgres is the same as the one created by `runCommand` in `/nix/store/...-pg-hba.conf`.

Reply:
I think it is enough to have a single line test to check if `hba_file` path starts with `/nix/store`